kernels/portable/cpu/util/elementwise_util.h

```diff
@@ -345,20 +346,22 @@ inline void apply_elementwise_fn(
   }
 
   constexpr auto compute_type = CppTypeToScalarType<CTYPE_COMPUTE>::value;
-  const bool all_inputs_compute_dtype =
-      ((inputs.first->scalar_type() == compute_type) && ...);
-
-  constexpr ScalarType out_specialized_scalar_type =
-      specialized_output_scalar_type<CTYPE_COMPUTE>(out_dtypes);
-  if (all_inputs_compute_dtype &&
-      out.scalar_type() == out_specialized_scalar_type) {
-    using CTYPE_OUT =
-        typename ScalarTypeToCppType<out_specialized_scalar_type>::type;
-    dtype_specialized_elementwise_fn_impl<
-        CTYPE_COMPUTE,
-        CTYPE_OUT,
-        support_noncontiguous_tensors>(compute_fun, ctx, out, inputs...);
-    return;
+  if constexpr (should_include_kernel_dtype(op_name, compute_type)) {
```

Comment:
I think I got this wrong. Talked to @lucylq and she confirmed that dtype selective build is about input and output tensor types, not the compute type. My plan is to fix this on main (see PR #12183 ) and to just revert this PR from the release branch for safety.